photon-client/src/main/scala/com/linkedin/photon/ml/cli/game/training/Driver.scala

```diff
@@ -167,21 +228,15 @@ final class Driver(val sc: SparkContext, val params: GameParams, implicit val lo
     }
 
   /**
-   * Check data, i.e. verify that it is admissible for training and/or validation.
-   *
-   * For now, we just check that the data contains at least a sample with non-zero weight). We throw out samples that
-   * have a weight less than or equal to 0. If we don't find at least one sample with a strictly positive weight, we
-   * throw an IllegalArgumentException, otherwise we return the filtered data set.
-   *
-   * @note this function has the side-effect of possibly reducing the size of the data set
-   * @note this is somewhat expensive (it iterates over the whole data), so the user can control it via parameter
-   *       checkData
+   * Check data, i.e. verify that it is admissible for training and/or validation. Currently, this checks that each
+   * sample weight has a strictly non-zero weight.
    *
+   * @note this is somewhat expensive (it iterates over the whole data), so the user can control it via the parameter
+   *       'checkData'
```

Comment:
Add a note that we have an explicit strategy of rejecting rather than fixing on behalf of others - so that this strategy might be maintained in the future.